src/memory_service/fam_memory_service_direct.cpp

```diff
@@ -133,12 +133,19 @@ Fam_Memory_Service_Direct::Fam_Memory_Service_Direct(uint64_t memserver_id,
 
     allocator =
         new Memserver_Allocator(num_delayed_free_Threads, fam_path.c_str());
-    fam_backup_path = (char *)strdup(config_options["fam_backup_path"].c_str());
+    std::string backup_location = config_options["fam_backup_path"];
+    std::size_t found = backup_location.find_last_of("/");
+    backup_location = backup_location.substr(0, found);
+    fam_backup_path = (char *)strdup(backup_location.c_str());
+
     struct stat info;
     if (stat(fam_backup_path.c_str(), &info) == -1) {
-        if ((errno == ENOENT) || (errno == ENOTDIR))
-            mkdir(fam_backup_path.c_str(),
-                  S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+        if ((errno == ENOENT) || (errno == ENOTDIR)) {
+            int ret = mkdir(fam_backup_path.c_str(),
+                            S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+            if (ret == -1)
+                cout << "Fam_Backup_Path Directory Creation failed. " << endl;
```

Comment:
@faizan-barmawer Should we return error here ? May be we can ignore this error during memory server startup and throw error during fam_backup.